Assets/Code/Scripts/Level generation scripts/TerrainGeneratorScript.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class TerrainGeneratorScript : MonoBehaviour
+{
+    private float tileWidth = 10.0f;
+
+    public int blocks = 2; //number of blocks being made, block x block
+    public int size = 5;
+    public float tavernPercent = 0.25f;
```

Comment:
Is there a reason for these to be public? If it's just for the editor, serialized is the way to go.

Reply:
I honestly have no idea what the difference is

Reply:
The difference is that a private [SerialisedFiled] is only tweakable in the editor, but a public field will be [SerialisedFiled] implicitly but also other classes will be able to acces and **modify** theses variables. If you don't want external classes to modify theses values at any time, don't put them public